### ISSUE TITLE
Store per-DApp reflection caches long-lived

### ISSUE THREAD

**Issue by aionbot** [aionnetwork/AVM#284]
**Issue created by [jeff-aion](https://github.com/jeff-aion)** (on Wednesday Oct 17, 2018 at 20:21 GMT)

Profiling while the hot DApp cache is active still resulted in reflection operations rising to the top.  These cases seemed to be due to `ReflectionStructureCodec` (when looking up constructors for stub instantiation, mostly) and `ABIDecoder` (when looking through method listings).

As these are only observable in relatively simple DApps, in the hot cache, and likely heavily JITted, this isn't a high priority but it is some relatively low-hanging fruit.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Monday Nov 19, 2018 at 14:43 GMT)

This is purely a performance item, and depends on #285, so it can be done fairly late.  That said, we should probably implement this sometime within the Beta timeframe since it should be considered within any performance analysis we do, leading up to TestNet.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Thursday Nov 29, 2018 at 19:46 GMT)

Moving to Future since this is low priority.
